polkadot/node/network/availability-distribution/src/requester/mod.rs

```diff
@@ -124,6 +161,57 @@ impl Requester {
 		Ok(())
 	}
 
+	async fn request_backable_candidates_core_info<Context>(
+		&mut self,
+		ctx: &mut Context,
+		new_head: ActivatedLeaf,
+	) -> Result<Vec<(CoreIndex, CoreInfo)>> {
+		let ActivatedLeaf { hash: leaf, .. } = new_head;
+		let mut scheduled_cores = Vec::new();
+		let sender = &mut ctx.sender().clone();
+
+		let cores = recv_runtime(request_availability_cores(leaf, sender).await).await?;
+		let backable = request_backable_candidates(&cores, None, &new_head, sender).await?;
+		gum::trace!(
+			target: LOG_TARGET,
+			leaf_hash=?leaf,
+			"Got {} backable candidates", backable.len()
+		);
+
+		// now get the backed candidates corresponding to these candidate receipts
+		let (tx, rx) = oneshot::channel();
+		sender
+			.send_message(CandidateBackingMessage::GetBackableCandidates(backable.clone(), tx))
+			.await;
+		let candidates = rx.await?;
+
+		// Process candidates and collect cores
+		for (_, candidates) in candidates.iter() {
+			for candidate in candidates {
+				let (_, core_index) = candidate.validator_indices_and_core_index();
```

Comment:
Sorry (bad API): This is not enough, with descriptor v2 we have the core index in the descriptor and no longer injected. We have to do something similarly to [here](https://github.com/chainsafe/polkadot-sdk/blob/da460f72e2f186a1a437d33b1d2bf553dddc07e1/polkadot/runtime/parachains/src/paras_inherent/mod.rs#L1548).

Reply:
Added the comment in https://github.com/paritytech/polkadot-sdk/pull/10563/commits/fb004b2e1c66c11a96a0854b0570316a8ee90b9d